StreamingKit/StreamingKit/STKHTTPDataSource.m

```diff
@@ -308,7 +308,10 @@ -(BOOL) parseHttpHeader
         self->iceHeaderData = nil;
     }
     
-    if (([httpHeaders objectForKey:@"Accept-Ranges"] ?: [httpHeaders objectForKey:@"accept-ranges"]) != nil)
+    NSString *accessControlAllowHeaders = ([httpHeaders objectForKey:@"Access-Control-Allow-Headers"] ?: [httpHeaders objectForKey:@"access-control-allow-headers"]);
+    if (([httpHeaders objectForKey:@"Accept-Ranges"] ?: [httpHeaders objectForKey:@"accept-ranges"]) != nil ||
+         (accessControlAllowHeaders != nil &&
+         ([accessControlAllowHeaders rangeOfString:@"Range"].location != NSNotFound || [accessControlAllowHeaders rangeOfString:@"range"].location != NSNotFound)))
```

Comment:
Could this potentially be condensed by only needing to check for a single `rangeOfString` against a lowercased `accessControlAllowHeaders`?